FpsCheck.py

```diff
@@ -103,13 +103,13 @@ def has_all_rationales(self, check_sets):
         error_list
 
         Args:
-            check_sets: a dictionary of primary->FpsSet
+            None
         Returns:
             None
         """
         for fpset in self.fps_sites['sets']:
-            sites = check_sets[fpset.get("primary")].associated_sites
-            service_sites = check_sets[fpset.get("primary")].service_sites
+            sites = fpset.get("associatedSites")
+            service_sites = fpset.get("serviceSites")
```

Comment:
Would this logic (and below) be simpler if we used a default value of `[]` for the associated sites and service sites, if the set doesn't have any? Then you could do:

```python
sites = fpset.get("associatedSites", []) + fpset.get("serviceSites", [])
# ...
if rationales:
  for site in sites:
    # ...
else:
  self.error_list.append(...)
```

Reply:
Implemented this in recent commit